railties/lib/rails/commands/console/console_command.rb

```diff
@@ -12,6 +12,47 @@ def filter_backtrace(bt)
       end
     end
 
+    class IRBConsole
+      def initialize
+        require "irb"
+        require "irb/completion"
+        IRB::WorkSpace.prepend(BacktraceCleaner)
+
+        if !Rails.env.local?
+          # Use env var here so users can override them with env var too
+          ENV["IRB_USE_AUTOCOMPLETE"] ||= "false"
+        end
+
+        IRB
+      end
+
+      def name
+        "IRB"
+      end
+
+      def start
+        IRB.setup(nil)
+
+        env =
```

Comment:
Could we colorize that env too? I'm thinking red in production, and yellow or green in dev/test?

Reply:
Done 👍

Reply:
@byroot are you attached to green in dev/test? If not, wdyt about using red/blue or red/yellow to make it distinguishable to folks with color blindness?

Reply:
I'm not attached to any particular color aside from `red` in production as `red` is commonly associated to danger. Any other color that is more distinguishable than green for color blind people is welcome.

Reply:
Here are the other options that `IRB::Color` provides. We can definitely use colours outside [ANSI escape sequences](https://en.wikipedia.org/wiki/ANSI_escape_code)' 8 colours, but then they'll need to be managed by Rails itself.

![Screenshot 2024-01-23 at 16 37 31](https://github.com/rails/rails/assets/5079556/21397384-1c8e-40c7-9165-3735488383cc)

Reply:
cc @kmcphillips has resident color blind specialist. What basic shell ANSI color to use with red for it to be accessible to most people?

Reply:
But also, just to note that red and green are not used together in the same prompt, they're used in different scenarios. So I think worst case it will be less obvious it's a production console for color blind people, but it should make it less readable.

Reply:
It's a thoughtful question to ask.

I don't have a great answer. For hex colour values you can look at a safe colour chart like this that takes the most common types of coloublindness into account:
https://en.wikipedia.org/wiki/Color_blindness#/media/File:Safe_Chart_Colors-F99-FEC-ADD.jpg

The real answer is: Do your best with colour but always provide a non-colour differentiator as a fallback.

Reply:
Here is the guideline I use when building anything that requires more than one color: https://www.datylon.com/blog/data-visualization-for-colorblind-readers#:~:text=The%20first%20rule%20of%20making,out%20of%20these%20two%20hues.

So I usually go for blue and red as they are, from what I understand, the safest combination for any type of color blindness. If blue is not possible, then yellow/red would be my next suggestion.

Reply:
Thanks for these great information, I learned a lot from them and the references.
I also have opened https://github.com/rails/rails/pull/50855 to change dev & test env's color to blue instead.